src/test/java/org/elasticsearch/percolator/PercolatorBackwardsCompatibilityTests.java

```diff
@@ -60,15 +63,30 @@ public void testPercolatorUpgrading() throws Exception {
                 .setSource(jsonBuilder().startObject().field("query", termQuery("field2", "value")).endObject()).get();
 
         // However on new indices, the field resolution is strict, no queries with unmapped fields are allowed
-        createIndex("test2");
+        createIndex("test1");
         try {
-            client().prepareIndex("test2", PercolatorService.TYPE_NAME)
+            client().prepareIndex("test1", PercolatorService.TYPE_NAME)
                     .setSource(jsonBuilder().startObject().field("query", termQuery("field1", "value")).endObject()).get();
             fail();
         } catch (PercolatorException e) {
             e.printStackTrace();
             assertThat(e.getRootCause(), instanceOf(QueryParsingException.class));
         }
-    }
 
+        // If and only if index.percolator.map_unmapped_fields_as_string is set to true, unmapped field in query is allowed.
+        ImmutableSettings.Builder settings = ImmutableSettings.settingsBuilder()
+                .put(indexSettings())
+                .put("index.percolator.map_unmapped_fields_as_string", true);
```

Comment:
The `index.percolator.map_unmapped_fields_as_string` is not a bwc thing. I think it is better to move the testing of this setting to the PercolatorTests class.